packages/hardhat/contracts/Auction.sol

```diff
@@ -1,132 +1,133 @@
-
-        // set bid status to cancled
-        // return deposit
-
-        bids[msg.sender] = _offer;
-        emit BidCancled(_offer, msg.sender);
-    }
+  using EnumerableSet for EnumerableSet.AddressSet;
+
+  struct AuctionType {
+    address manager; // created in SC - msg.sender
+    string title; // input
+    string description; // input
+    uint256 maxOffer; // input
+    uint256 submissionDeadline; // input
+    uint256 startDate; // created in SC
+    uint256 endDate; // created in SC
+  }
+
+  enum AuctionStatus {
+    Opened,
+    Closed,
+    Canceled
+  }
+
+  enum BidStatus {
+    Accepted,
+    Declined,
+    Placed,
+    Canceled
+  }
+
+  struct BidType {
+    uint256 offerAmount; // input
+    string description; // input
+    BidStatus status; // accepted or not - // created/set in SC
+  }
+
+  AuctionType auctionInfo;
+  AuctionStatus auctionStatus;
+
+  address public winningBidder;
+  uint256 public winningBid;
+
+  EnumerableSet.AddressSet private bidders;
+  mapping(address => BidType) public biddingForAuction;
+
+  uint256 public maxBidOffer;
+
+  event BidPlaced(uint256 offer, string description, address indexed bidder);
+  event BidCanceled(uint256 offer, address indexed bidder);
+  event AuctionFinalized(address indexed manager, string title, uint256 maxOffer, uint256 endDate, address indexed bidder, uint256 winningBid);
+  event AuctionCanceled(address indexed manager, string title, uint256 maxOffer, uint256 endDate, address indexed bidder, uint256 winningBid);
+
+  constructor(
+    address _manager,
+    string memory _title,
+    string memory _description,
+    uint256 _maxOffer,
+    uint256 _submissionDeadline,
+    uint256 _startDate,
+    uint256 _endDate
+  ) {
+    auctionInfo = AuctionType(_manager, _title, _description, _maxOffer, _submissionDeadline, _startDate, _endDate);
+    auctionStatus = AuctionStatus.Opened;
+    maxBidOffer = _maxOffer + 1;
+  }
+
+  modifier onlyManager() {
+    require(auctionInfo.manager == msg.sender, 'Caller is not manager');
+    _;
+  }
+
+  function getBidders() external view returns (address[] memory) {
+    return bidders.values();
+  }
+
+  function finalizeAuction(address _winningBidder, uint256 _winningBid) public onlyManager {
+    require(auctionStatus == AuctionStatus.Opened, 'Auction has already been finalized');
+    require(biddingForAuction[_winningBidder].status != BidStatus.Canceled, 'Canceled Bid');
+    // require(block.timestamp >= endDate, "Auction is still active.");
+
+    winningBidder = _winningBidder;
+    winningBid = _winningBid;
+
+    // Perform finalization actions, e.g., transfer assets to the winner.
+    // manager has to deposit the funds into the conttract - amount = _winningBid
+    // status of all Biddings has to be set to declined except the choosen one
+    // all deposits of all people that places a bid is being returned IMPORTANT
+    // set auction status to closed
```

Comment:
can you update the comments here to:

please remove all comments that are not valid anymore

// TODO: add logic for payment transfer. Blocked by finalization of WorkContract.sol

Regarding the status of the Biddings:
We have to at least set the status of the winning bid to `accepted`. 
If it will be too much work to set everyone else to `declined`, we can leave them on `placed`.  If not, please set them to declined.